app/code/Magento/SalesRule/Model/Rule/Condition/Product/Combine.php

```diff
@@ -6,32 +6,37 @@
 namespace Magento\SalesRule\Model\Rule\Condition\Product;
 
 use Magento\Catalog\Model\ResourceModel\Product\Collection;
+use Magento\Framework\Model\AbstractModel;
+use Magento\Rule\Model\Condition\Combine as ConditionCombine;
+use Magento\Rule\Model\Condition\Context as ConditionContext;
+use Magento\SalesRule\Model\Rule\Condition\Product as RuleCondProduct;
+use Magento\SalesRule\Model\Rule\Condition\Product\Combine as RuleCondProductCombine;
 
 /**
  * Combine conditions for product.
  * @api
  * @since 100.0.2
  */
-class Combine extends \Magento\Rule\Model\Condition\Combine
+class Combine extends ConditionCombine
 {
     /**
-     * @var \Magento\SalesRule\Model\Rule\Condition\Product
+     * @var RuleCondProduct
      */
     protected $_ruleConditionProd;
 
     /**
-     * @param \Magento\Rule\Model\Condition\Context $context
-     * @param \Magento\SalesRule\Model\Rule\Condition\Product $ruleConditionProduct
+     * @param ConditionContext $context
+     * @param RuleCondProduct $ruleConditionProduct
      * @param array $data
      */
     public function __construct(
-        \Magento\Rule\Model\Condition\Context $context,
-        \Magento\SalesRule\Model\Rule\Condition\Product $ruleConditionProduct,
+        ConditionContext $context,
+        RuleCondProduct $ruleConditionProduct,
```

Comment:
Why not promote this property?
```suggestion
        protected readonly RuleCondProduct $_ruleConditionProd,
```